home-manager/home-manager

```diff
@@ -100,6 +136,47 @@ function doListPackages() {
     fi
 }
 
+function doBuildNews() {
+    local dataDir="${XDG_DATA_HOME:-$HOME/.local/share}/home-manager"
+    local timestampPath="$dataDir/news-timestamp"
+    local since="1970-01-01T00:00:00+00:00"
+
+    if [[ -r "$timestampPath" ]]; then
+        read -r since < "$timestampPath"
+    fi
+
+    doBuildAttr "$*" \
+                --no-out-link \
+                --arg check false \
+                --argstr newsSince "$since"
+}
+
+function doShowNews() {
+    local infoFile
+    infoFile=$(doBuildNews -A newsInfo) || return 1
+
+    # shellcheck source=/dev/null
+    . "$infoFile"
+
+    case $1 in
+        --all)
+            ${PAGER:-less} "$newsFileAll"
+            ;;
+        --unread)
+            ${PAGER:-less} "$newsFileUnread"
+            ;;
+        *)
+            echo "Unknown argument $1"
+            return 1
+    esac
+
+    local dataDir="${XDG_DATA_HOME:-$HOME/.local/share}/home-manager"
+    local timestampPath="$dataDir/news-timestamp"
```

Comment:
`dataDir` and `timestampPath` are defined twice with the same values. Maybe make them global?

Reply:
Yeah, I'll change it to avoid the code duplication.